diff -Naur repo_orig/test_image.py repo/test_image.py
--- repo_orig/test_image.py	Wed Feb 14 17:26:44 2024
+++ repo/test_image.py	Wed Feb 14 17:31:48 2024
@@ -44,11 +44,29 @@
     img, _, _, _ = transformer(img, None, torch.zeros(1,4), torch.zeros(1))
     encoder = Encoder(dboxes)
 
+    to_export_model = os.environ.get('CM_ABTF_EXPORT_MODEL_TO_ONNX','')
+    exported = False
+    
     if device=='cuda':
         img = img.cuda()
     with torch.no_grad():
-        ploc, plabel = model(img.unsqueeze(dim=0))
+        inp = img.unsqueeze(dim=0)
+        
+        ploc, plabel = model(inp)
         result = encoder.decode_batch(ploc, plabel, opt.nms_threshold, 20)[0]
+
+        if to_export_model!='' and not exported:
+            torch.onnx.export(model,
+                 inp,
+                 to_export_model,
+                 verbose=True,
+                 input_names=['input'],
+                 output_names=['output'],
+                 export_params=True,
+                 )
+
+            exported = True     
+
         loc, label, prob = [r.cpu().numpy() for r in result]
         best = np.argwhere(prob > opt.cls_threshold).squeeze(axis=1)
         loc = loc[best]
