diff --git a/vision/classification_and_detection/tools/accuracy-openimages.py b/vision/classification_and_detection/tools/accuracy-openimages.py
index 0192dde..7fb0dd6 100644
--- a/vision/classification_and_detection/tools/accuracy-openimages.py
+++ b/vision/classification_and_detection/tools/accuracy-openimages.py
@@ -34,7 +34,7 @@ def get_args():
 def main():
     args = get_args()
 
-    cocoGt = COCO(os.path.join(args.openimages_dir, "annotations/openimages-mlperf.json"))
+    cocoGt = COCO(os.path.join(args.openimages_dir, "annotations/openimages-mlperf.json"), use_ext=True)
 
     if args.use_inv_map:
         inv_map = [0] + cocoGt.getCatIds() # First label in inv_map is not used
@@ -98,8 +98,8 @@ def main():
     with open(args.output_file, "w") as fp:
         json.dump(detections, fp, sort_keys=True, indent=4)
 
-    cocoDt = cocoGt.loadRes(args.output_file) # Load from file to bypass error with Python3
-    cocoEval = COCOeval(cocoGt, cocoDt, iouType='bbox')
+    cocoDt = cocoGt.loadRes(args.output_file,use_ext=True) # Load from file to bypass error with Python3
+    cocoEval = COCOeval(cocoGt, cocoDt, iouType='bbox',use_ext=True)
     cocoEval.params.imgIds = list(image_ids)
     cocoEval.evaluate()
     cocoEval.accumulate()
